Penteract/Assets/Scripts/PlayerController.cpp

```diff
@@ -245,17 +190,80 @@ void PlayerController::SwitchCharacter() {
 			hudControllerScript->UpdateHP(lifePointsFang, lifePointsOni);
 		}
 		switchCooldownRemaing = switchCooldown;
+		if(hudControllerScript){
+			hudControllerScript->ChangePlayerHUD();
+		}
 	}
 }
 
-void PlayerController::ReceiveEvent(TesseractEvent& e) {
+bool PlayerController::CanShoot() {
+	return !shooting && ((fang && fangCompParticle) || (onimaru && onimaruCompParticle));
+}
+
+void PlayerController::Shoot() {
+	if (CanShoot()) {
+		if (shootAudioSource) {
+			shootAudioSource->Play();
+		}
+		else {
+			Debug::Log(AUDIOSOURCE_NULL_MSG);
+		}
+
+		shootCooldownRemaing = shootCooldown;
+		shooting = true;
+		if (fang->IsActive()) {
+			fangCompParticle->Play();
+		}
+		else {
+			onimaruCompParticle->Play();
+		}
 
+		float3 start = transform->GetPosition();
```

Comment:
Maybe here you want the global position?

Reply:
yeah it is correct, i will fix it